File: dotnet/niuz.application/commands/CommandDispatcher.cs
using System.Linq;

namespace niuz.application.commands
{
    public class CommandDispatcher
    {
        private readonly object[] handlers;

        public CommandDispatcher(params object[] handlers)
        {
            this.handlers = handlers;
        }

        public void Dispatch<T>(T command)
        {
            handlers
                .OfType<ICommandHandler<T>>()
                .First()
                .Handle(command);
            
        }
    }
}

File: java/src/main/java/com/dddeurope/niuz/commands/CommandDispatcher.java
package com.dddeurope.niuz.commands;

import com.dddeurope.niuz.hr.HireAuthor;

import java.util.Arrays;
import java.util.List;

public class CommandDispatcher {
    private final List<CommandHandler> handlers;

    public CommandDispatcher(CommandHandler<HireAuthor>... handlers) {
        this.handlers = Arrays.asList(handlers);
    }

    public void dispatch(Object command) {
        handlers.stream()
                .filter(handler -> handler.canHandle(command))
                .forEach(handler -> handler.handle(command));
    }
}
